File: LeetCode/0256.Paint-House/Paint-House.java
class Solution {
    public int minCost(int[][] costs) {
        if(costs == null || costs.length == 0 || costs[0].length == 0) return 0;
        for(int i = 1; i < costs.length; i++){
            costs[i][0] += Math.min(costs[i - 1][1], costs[i - 1][2]);
            costs[i][1] += Math.min(costs[i - 1][0], costs[i - 1][2]);
            costs[i][2] += Math.min(costs[i - 1][1], costs[i - 1][0]);
        }
        int m = costs.length - 1;
        return Math.min(costs[m][0], Math.min(costs[m][1], costs[m][2]));
    }
}

File: LeetCode/0256.Paint-House/Paint-House.py
class Solution(object):
    def minCost(self, costs):
        """
        :type costs: List[List[int]]
        :rtype: int
        """
        if not costs: return 0
        res = list(costs[0])
        n = len(costs)
        for i in range(1, n):
            cur = list(costs[i])
            cur[0] += min(res[1], res[2])
            cur[1] += min(res[0], res[2])
            cur[2] += min(res[0], res[1])
            res = list(cur)
        return min(res)
